packages/edit-post/src/components/preferences-modal/index.js

```diff
@@ -44,6 +52,22 @@ import BlockManager from '../block-manager';
 const MODAL_NAME = 'edit-post/preferences';
 const PREFERENCES_MENU = 'preferences-menu';
 
+function NavigationButton( {
+	as: Tag = Button,
```

Comment:
I tried to make this general enough but because of the `isAction` of the Item component, this was not enough. So I was wondering if we should remove the `isAction` prop from the `Item` component. Would be enough to infer it from the presence of `onClick`?

Reply:
> So I was wondering if we should remove the isAction prop from the Item component. Would be enough to infer it from the presence of onClick?

While I'm not necessarily a fan of the `isAction` prop on the `Item` component, I'm not sure that relying only on the `onClick` prop is the best solution here. Currently, when `isAction` is `true`, [the `Item` component renders as a `Button` _and_ applies some extra styles (to undo some of the `Button` look&feel)](https://github.com/WordPress/gutenberg/blob/92662feeb220336fb28f8b79869cb251a7db4e27/packages/components/src/item-group/item/hook.ts).

According to your approach, what if we specified the `onClick` prop, but we wanted to render the component `as` something different than a `@wordpress/components`'s `Button` ?

--- 

In case we're keeping `isAction` for this PR, we may want to modify the code slightly so that we pass the `isAction` prop only when we're rendering `NavigationButton` as an `Item`:

```diff
diff --git a/packages/edit-post/src/components/preferences-modal/index.js b/packages/edit-post/src/components/preferences-modal/index.js
index 468ea0bca7..2fd66fa44f 100644
--- a/packages/edit-post/src/components/preferences-modal/index.js
+++ b/packages/edit-post/src/components/preferences-modal/index.js
@@ -61,7 +61,6 @@ function NavigationButton( {
 	const navigator = useNavigator();
 	return (
 		<Tag
-			isAction
 			onClick={ () => navigator.push( path, { isBack } ) }
 			{ ...props }
 		/>
@@ -336,6 +335,7 @@ export default function PreferencesModal() {
 											key={ tab.name }
 											path={ tab.name }
 											as={ Item }
+											isAction
 										>
 											<HStack justify="space-between">
 												<FlexItem>
```

Reply:
> According to your approach, what if we specified the onClick prop, but we wanted to render the component as something different than a @wordpress/components's Button ?

Then for me, `Button` should be used when onClick is defined but no `as`

Reply:
I'll look into it and open a draft PR to explore this scenario

Reply:
Open #35152 to try this approach

Reply:
With #35152 merged, you may want to rebase this PR and remove this `isAction` prop from all usages of `Item`

Reply:
Do we need to make this general as all are usages are `buttons`? Also `isBack` property needed?

Reply:
I'm using `Item` sometimes instead of `Button`